### PR TITLE
Update Helm docs to reflect 0.42.0 release

### PR DESCRIPTION
**Pull request by t-eckert**
Update Helm docs to reflect the v0.42.0 release.

### CODE REVIEW

File: website/content/docs/k8s/helm.mdx [hunk 1]
```diff
@@ -408,11 +440,25 @@ Use these links to navigate to a particular top-level stanza.
...
-    - `primaryDatacenter` ((#v-global-federation-primarydatacenter)) (`string: ""`) - The name of the primary datacenter.
+    - `primaryDatacenter` ((#v-global-federation-primarydatacenter)) (`string: null`) - The name of the primary datacenter.
 
     - `primaryGateways` ((#v-global-federation-primarygateways)) (`array<string>: []`) - A list of addresses of the primary mesh gateways in the form `<ip>:<port>`.
       (e.g. ["1.1.1.1:443", "2.3.4.5:443"]
 
+    - `k8sAuthMethodHost` ((#v-global-federation-k8sauthmethodhost)) (`string: null`) - If you are setting `global.federation.enabled` to true and are in a secondary datacenter,
+      set `k8sAuthMethodHost` to the address of the Kubernetes API server of the secondary datacenter.
+      This address must be reachable from the Consul servers in the primary datacenter.
+      This authmethod will be used to provision ACL tokens for Consul components and is different
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/helm.mdx:451]:
```suggestion
      This auth method will be used to provision ACL tokens for Consul components and is different
```

File: website/content/docs/k8s/helm.mdx [hunk 2]
```diff
@@ -408,11 +440,25 @@ Use these links to navigate to a particular top-level stanza.
...
+      from the one used by the Consul Service Mesh.
+      Please see the Kubernetes Auth Method documentation (https://consul.io/docs/acl/auth-methods/kubernetes).
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/helm.mdx:453]:
```suggestion
      Please see the [Kubernetes Auth Method documentation](https://consul.io/docs/acl/auth-methods/kubernetes) for additional information.
```

File: website/content/docs/k8s/helm.mdx [hunk 3]
```diff
@@ -408,11 +440,25 @@ Use these links to navigate to a particular top-level stanza.
...
+
+      You could retrieve this value from your `kubeconfig` by running:
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/helm.mdx:455]:
```suggestion
      You can retrieve this value from your `kubeconfig` by running:
```

**Review by karl-cardenas-coding — APPROVED** [review]
👍🏼

**Comment by david-yu** [comment]
Resolved comments here: https://github.com/hashicorp/consul-k8s/pull/1142 as well

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/623719.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit f51c02a9237372be29a79bfa1c47ab5060dc235a onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit f51c02a9237372be29a79bfa1c47ab5060dc235a onto `release/1.12.x` succeeded!

**Pull request merged**